src/image_tools/bake.py

```diff
@@ -159,18 +156,71 @@ def bake_command(args: Namespace, product_name: str, bakefile) -> Command:
     )
 
 
+def filter_product_version(conf, product_name: Optional[str], product_version: Optional[str],
+                           product_version_quantile: Optional[int], max_bake_runners: int) -> Optional[int]:
+    """ Filter product versions by the given product_version argument.
+        Mutates the "conf.products" array to remove version of "product_name" that don't match "product_version"
```

Comment:
Why not just return the new list instead?

Reply:
I thought about it but it couldn't see the advantage. It would have to the list for one particular product (the one to build) and the only use-case would be to check for it's length.

As write this, it occurs to me that we could also raise an error if there is nothing to build. Would you agree to that?

Reply:
Honestly, it depends.. "your request matched 0 products" is definitely an error condition in my book, "your shard contains 0 builds" is not.

Reply:
Yes, I agree. The error we raise here would not propagate as an error to the user.

And yes, a check if the product actually exists should be added and should result in a user facing error.